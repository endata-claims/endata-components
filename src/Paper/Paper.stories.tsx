import React from 'react'

import Paper from '.'

export default {
  title: 'Paper',
}

export const Default = () => {
  return (
    <Paper title='Title'>Content</Paper>
  )
}

export const Loading = () => {
  return (
    <Paper title='Title' loading={true}>
      Content
    </Paper>
  )
}