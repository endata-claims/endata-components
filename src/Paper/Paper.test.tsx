import React from 'react'
import { shallow } from 'enzyme'

import Paper from '.'

describe('Paper tests', () => {
  const renderWrapper = (props = {}) => shallow(<Paper {...props} />)

  describe('snapshots', () => {
    it('should render without crash', () => {
      expect(renderWrapper()).toMatchSnapshot()
    })
    it('should render loading', () => {
      expect(renderWrapper({ loading: true })).toMatchSnapshot()
    })
    it('should render with title', () => {
      expect(renderWrapper({ title: 'Paper title' })).toMatchSnapshot()
    })
  })
})