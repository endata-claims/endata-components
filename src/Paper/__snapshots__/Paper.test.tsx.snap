// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper tests snapshots should render loading 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paperContent-2"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(CircularProgress)) />
    </div>
  </WithStyles(ForwardRef(Paper))>
</Fragment>
`;

exports[`Paper tests snapshots should render with title 1`] = `
<Fragment>
  <div
    className="makeStyles-paperTitle-1"
  >
    Paper title
  </div>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paperContent-2"
  />
</Fragment>
`;

exports[`Paper tests snapshots should render without crash 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paperContent-2"
  />
</Fragment>
`;
